ios-itunes/ios-itunes/Views/SearchResultsTableViewController.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  SearchResultsTableViewController.swift
+//  ios-itunes
+//
+//  Created by Angel Buenrostro on 1/22/19.
+//  Copyright © 2019 Angel Buenrostro. All rights reserved.
+//
+
+import UIKit
+
+class SearchResultsTableViewController: UITableViewController, UISearchBarDelegate {
+
+    @IBOutlet weak var controlBar: UISegmentedControl!
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    let searchResultsController = SearchResultController()
+    
+    
+    func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        guard let searchTerm = searchBar.text, !searchTerm.isEmpty else { return }
+        
+        var resultType : ResultType!
+        switch controlBar.selectedSegmentIndex {
+        case 0:
+            resultType = .software
+        case 1:
+            resultType = .musicTrack
+        case 2:
+            resultType = .movie
+        default:
+            print("Error setting the appropriate resultType")
+        }
+        
+        searchResultsController.performSearch(searchTerm: searchTerm, resultType: resultType) { (error) in
+            DispatchQueue.main.async {
```

Comment:
After getting results back. You could check the error `(if let error = error)` and log it if there is something else. Then move on to `DispatchQueue`